mpqp/core/instruction/measurement/basis.py

```diff
@@ -96,17 +137,62 @@ def pretty_print(self):
         print(f"Basis: [\n    {joint_vectors}\n]")
 
     def __repr__(self) -> str:
-        return f"{type(self).__name__}({self.basis_vectors}, {self.nb_qubits})"
+        joint_vectors = ", ".join(map(one_lined_repr, self.basis_vectors))
+        qubits = "" if isinstance(self, VariableSizeBasis) else f", {self.nb_qubits}"
+        return f"{type(self).__name__}({joint_vectors}{qubits})"
+
+    def to_computational(self):
+        """Converts the custom basis to the computational basis.
+
+        This method creates a quantum circuit with a custom gate represented by
+        a unitary transformation and applies it to all qubits before measurement.
+
+        Returns:
+            A quantum circuit representing the basis change circuit.
+
+        Example:
+            >>> basis = Basis([np.array([1, 0]), np.array([0, -1])])
+            >>> circuit = basis.to_computational()
+            >>> print(circuit)
+               ┌─────────┐
+            q: ┤ Unitary ├
+               └─────────┘
+
+        """
+
+        from mpqp.core.circuit import QCircuit
+
+        basis_change = np.array(self.basis_vectors).T.conjugate()
+        return QCircuit(
+            [
+                CustomGate(
+                    UnitaryMatrix(basis_change), targets=list(range(self.nb_qubits))
+                )
+            ]
+        )
 
 
 @typechecked
-class VariableSizeBasis(Basis):
-    """3M-TODO"""
+class VariableSizeBasis(Basis, ABC):
+    """A variable-size basis with a dynamically adjustable size to different qubit numbers
+    during circuit execution.
+
+    Args:
+        nb_qubits: number of qubits in the basis. If not provided,
+            the basis can be dynamically sized later using the `set_size` method.
+
+        symbols: custom symbols for representing basis states, defaults to ("0", "1").
+    """
 
     @abstractmethod
-    def __init__(self, nb_qubits: Optional[int] = None):
-        super().__init__([], nb_qubits)
-        pass
+    def __init__(
+        self, nb_qubits: Optional[int] = None, symbols: Optional[tuple[str, str]] = None
+    ):
+        if symbols is None:
```

Comment:
This is done already in the `super().__init__(...)`, so maybe we can remove those lines ?

Reply:
agreed